service/history/task/processor.go

```diff
@@ -22,70 +22,86 @@ package task
 
 import (
 	"errors"
+	"fmt"
 	"sync"
 	"sync/atomic"
 
 	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/service/dynamicconfig"
 	"github.com/uber/cadence/common/task"
+	"github.com/uber/cadence/service/history/config"
 	"github.com/uber/cadence/service/history/shard"
 )
 
 type (
-	// ProcessorOptions configs task processor
-	ProcessorOptions struct {
-		SchedulerType        task.SchedulerType
-		FifoSchedulerOptions *task.FIFOTaskSchedulerOptions
-		WRRSchedulerOptions  *task.WeightedRoundRobinTaskSchedulerOptions
+	schedulerOptions struct {
+		schedulerType        task.SchedulerType
+		fifoSchedulerOptions *task.FIFOTaskSchedulerOptions
+		wrrSchedulerOptions  *task.WeightedRoundRobinTaskSchedulerOptions
 	}
 
 	processorImpl struct {
 		sync.RWMutex
 
 		priorityAssigner PriorityAssigner
-		schedulers       map[shard.Context]task.Scheduler
+		scheduler        task.Scheduler
+		shardSchedulers  map[shard.Context]task.Scheduler
```

Comment:
What about rename to schedulerByHost and shecudlersByShards? Would it be easier to understand?

Reply:
Updated `scheduler` -> `hostScheduler`